### PR TITLE
[ROX-19482] : Add data model changes, conversion and validation support for custom email subject and body

### PR DESCRIPTION
**Pull request by charmik-redhat**
- Added fields to support custom email subject and body in the storage and API data models.
- Added validations for the custom subject and body when creating/updating report configuration.
- Updated conversions between storage and API models to convert custom subject and body.

## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [X] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.
This change does not require any special upgrade steps. Change log and documentation will be added in a separate PR.

### Here I tell how I validated my change

TODO(replace-me)
Unit tests

Manual tests by making API calls to 
1) create a report configuration with custom subject and body, 
2) getting the report configuration and verifying that the custom subject and body are set.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/reports/service/v2/convert.go [hunk 1]
```diff
@@ -133,10 +133,13 @@ func (s *serviceImpl) convertV2NotifierConfigToProto(notifier *apiV2.NotifierCon
 			Id: notifier.GetEmailConfig().GetNotifierId(),
 		},
 	}
-	if notifier.GetEmailConfig() != nil {
+
+	if emailConfig := notifier.GetEmailConfig(); emailConfig != nil {
 		ret.NotifierConfig = &storage.NotifierConfiguration_EmailConfig{
 			EmailConfig: &storage.EmailNotifierConfiguration{
-				MailingLists: notifier.GetEmailConfig().GetMailingLists(),
```
Comment (ksurabhi91) [inline central/reports/service/v2/convert.go:139]: Add notifier id

Comment (charmik-redhat) [reply central/reports/service/v2/convert.go:139]: For storage proto, we do not store notifier id in the `EmailNotifierConfiguration`, but we added a separate `one of` field called `Ref` inside `NotifierConfiguration`. The notifier id is assigned to `Ref` on ln133 above.

File: generated/api/v1/report_configuration_service.swagger.json [hunk 2]
```diff
@@ -457,6 +457,12 @@
           "items": {
             "type": "string"
           }
+        },
+        "customSubject": {
```
Comment (theencee) [inline generated/api/v1/report_configuration_service.swagger.json:461]: why is the v1 api being changed?

**Review by theencee — APPROVED** [review]

File: generated/api/v1/report_configuration_service.swagger.json (hunk 2 above)
Comment (charmik-redhat) [reply generated/api/v1/report_configuration_service.swagger.json:461]: The v1 api is the same as storage proto. So, any changes that happen to storage proto also get documented in v1 api. There is a `hidden` option but that is only for the search tags I think.

Comment (ksurabhi91) [reply generated/api/v1/report_configuration_service.swagger.json:461]: I had the same question

**Review by ksurabhi91 — APPROVED** [review]

**Review by ksurabhi91 — COMMENTED** [review]
Fix all style check failures in CI

**Review by md2119 — APPROVED** [review]

**Pull request merged**